src/Illuminate/Support/Str.php

```diff
@@ -818,6 +818,17 @@ public static function ucfirst($string)
         return static::upper(static::substr($string, 0, 1)).static::substr($string, 1);
     }
 
+    /**
+     * Encodes the string to a URL-query-safe string.
+     *
+     * @param  $string
+     * @return string
+     */
+    public static function urlQuery($string)
+    {
+        return urlencode(static::lower($string));
```

Comment:
Why lowercase? Seems opinionated

Reply:
You're totally right, shouldn't lower case, but may be I should add the option to normalize the string to lowercase.